Update version-5.3.md (#4343)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-dashboard/release-notes/version-5.3.md

#### Tyk Cloud
Tyk 5.3.0 will be available on Tyk Cloud in the coming weeks. We'll update this page once it's ready.

#### Self-Managed
This release is ready for installation on your own infrastructure.

### Breaking Changes

**Attention: Please read this section carefully.**

#### Tyk OAS APIs are not backward compatible with pre-5.3.0 Tyk OAS API Definitions

> This upgrade transitions Tyk OAS APIs out of [Early Access]({{< ref "frequently-asked-questions/using-early-access-features" >}}).

Reviewer: ```suggestion
This upgrade transitions Tyk OAS out of [Early Access]({{< ref "frequently-asked-questions/using-early-access-features" >}}).
```

This sentence should say _Tyk OAS_, not _Tyk OAS APIs_.

Author: @andyo-tyk can you please explain

Reviewer: The "OAS" concept within Tyk Dashboard and Tyk Gateway is called "Tyk OAS".
**This** is what is transitioning out of Early Access in 5.3.0.
Tyk OAS APIs are APIs that use the Tyk OAS API Definition which is the Tyk OAS way of configuring Tyk.

- Historic Tyk OAS API Definition Compatibility: 
  - Tyk OAS API Definitions prior to version 5.3.0 are not compatible with Tyk Gateway v5.3.X.

>   - This means that the TYk OAS API definitions you created in any previous release (5.1.0-5.2.x) cannot work with the new Tyk Dashboard without migrating them to the new format.

Reviewer: ```suggestion
  - This means that any Tyk OAS APIs created in any previous release (4.1.0-5.2.x) cannot work with the new Tyk Dashboard without being migrated to the new format.
```

Corrected a typo, improved clarity. Not sure why we're explicitly stating releases 5.1.0-5.2.x, given that Tyk OAS APIs have been around since 4.1.0, so have altered the version range.

Author: because I made a mistake ;)

- After upgrade: 

>   - If you had a Tyk OAS API Definition prior to version 5.3.0 then the new Tyk Dashboard will automatically update for you all your definitions to use the [new format]({{< ref "tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc" >}}).

Reviewer: ```suggestion
  - If you had a Tyk OAS API prior to version 5.3.0 then Tyk Dashboard will automatically update the API definition to [latest format]({{< ref "tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc" >}}).
```

>   - This means that you should not be worried about making your Tyk OAS API definitions compatible with the new 5.3.0 release as Tyk Dashboard will take care of that during start-up.

Reviewer: ```suggestion
  - This means that you do not have to do anything to make your Tyk OAS APIs compatible with the new 5.3.0 release as Tyk Dashboard will take care of that during start-up.
```

> - Non-Reversible Change:

Reviewer 2: See comment made by @lghiur in [GW release notes PR](https://github.com/TykTechnologies/tyk-docs/pull/4038#discussion_r1551394451). @lghuir does the same apply here

Reviewer 3: this looks accurate for Cloud

>   - The upgrade is not reversible (i.e. you cannot use the Tyk OAS API definitions with an older Tyk Dashboard).

Reviewer: ```suggestion
  - The upgrade is not reversible (i.e. you cannot use version 5.3.0-compatible Tyk OAS API definitions with an older Tyk Dashboard).
```

>   - This means rollback or reverting to your previous version requires restoring these API definitions from a backup.

Reviewer: ```suggestion
  - This means that to rollback or revert to your previous version of Tyk, you will need to restore these API definitions from a backup.
```

##### Tyk OAS API - Gateway Compatibility

> - If you are using Tyk OAS APIs, Tyk Dashboard 5.3.0 would only work with Tyk Gateway v5.3.0 and MDCB v2.5. Older versions of Tyk Gateway may not function correctly with the updated Tyk OAS API definitions. 

Reviewer: ```suggestion
- If you are using Tyk OAS APIs, Tyk Dashboard 5.3.0 will only work with Tyk Gateway v5.3.0 and MDCB v2.5 or later. Older versions of Tyk Gateway may not function correctly with version-5.3.0-compatible Tyk OAS API definitions. 
```

<!-- ##### Future deprecations
-->

### Upgrade instructions for 5.3.0

#### Essential steps before upgrade

1. Back-Up Your Environment (this is the [usual guidance]({{<ref "upgrading-tyk#tyk-self-managed">}}) with every release)

> 2. Export Tyk OAS API Definitions (this step is specific to this release due to being in the early access stage till now and not committing to a consistent Tyk OAS API format. The dashboard migrated and updated the Tyk OAS API definitions in the database and as such this upgrade is not reversible. 

Reviewer: ```suggestion
2. Export Tyk OAS API Definitions. With Tyk OAS leaving Early Access in this release, the Dashboard will migrate your Tyk OAS APIs in the database to the version-5.3.0-compatible API definition at which point they will no-longer work with pre-5.3.0 Tyk Dashboard. We advise you to backup your Tyk OAS API definitions before upgrading Tyk Dashboard in case you later need to rollback to an earlier Tyk version 
```

I'm not sure why this isn't included in the "usual guidance" (taking a back up of your APIs)

Author: that's not a question for a discussion atm. We never touched the API defs till now.

#### Upgrade process

> <Add instructions for 5.3.0!!!>

Reviewer 2: Do we need the upgrade process instructions for 5.3.0? Confirmed in scrum of scrum meet that order of upgrading components does not matter.

Reviewer 3: @dcs3spp they do need to backup OAS definitions right? in case they later decide to downgrade

Reviewer: Yes, I recommend that here we reinforce the (standard) advice to backup all API definitions before upgrading Tyk - but then link to the standard upgrade docs.

Reviewer: Actually, on reflection - we've given very clear instruction immediately before this section that they should backup the API defs before upgrade, so perhaps we don't need to state it again here.

Author: agree

> <Add instructions for 5.3.0!!!>

Author: ```suggestion
```

> Users are strongly advised to review the recommendation in the [upgrade guide](#upgrading-tyk) before applying any updates.

Reviewer: ```suggestion
Users are strongly advised to follow the instructions in the [upgrade guide](#upgrading-tyk) when upgrading Tyk.
```
